### DISCUSSION TITLE
Option to disable all schema and RLS editing through the UI

### DISCUSSION THREAD

**Discussion by icopp** [supabase/supabase#16759] [Feature Requests]
For any project trying to maintain a database through migrations, it's kind of an nightmare that the UI actively encourages changing so much in place. What's needed is some kind of project-level setting that disables any editing of tables, triggers, RLS policies, and anything else that can be done through SQL to enforce only making changes via migrations.

**Comment by MildTomato** [comment]
This is a part duplicate of https://github.com/orgs/supabase/discussions/12432, but we'll keep both open since they both have 1 upvote and cover different use cases.

**Comment by akashagarwal7** [comment]
+1
